draft-seemann-quic-nat-traversal.md

```diff
@@ -178,6 +178,39 @@ number of concurrent path validations.
 Endpoints SHOULD set an active_connection_id_limit that is high enough to allow
 for the desired number of concurrent path validation attempts.
 
+### Probing Paths when using QUIC Multipath
+
+If the QUIC Multipath extensions have been negotiated (see {{MULTIPATH}}),
+the probing of paths will be as described in {{probing-paths}}, but the
+client should send the candidate pairs using PUNCH_MY_PATH frames instead
+of PUNCH_ME_NOW frames. The PUNCH_MY_PATH frames carry the same information
+as the PUNCH_ME_NOW frame, plus a path identifier (Path ID), chosen
+from the range of valid Path ID. The server should
+use connection identifiers associated to that Path-ID when validating
+the path.
+
+If the selected Path ID is not yet in use, successful path probing will
+result in the creation of a new path, to be used in combination with the
+pre-existing paths.
+
+If the selected Path ID is already in use, successful path probing will
+result in the migration of the existing path to a new address.
```

Comment:
Just to make sure I understand, we're saying: 
- If this is a new path, it just brings up a new path, potentially having hole-punched it
- If it's an existing path, please hole-punch it over here, like if we did PUNCH_ME_NOW, but since it's MP-QUIC, just move that path and not everything else

Reply:
I rewrote that part, because the Multipath logic will not normally do path migration. Having a packet arriving with a new tuple on an existing path signals a NAT rebinding. That will trigger a Path Challenge on old and new path, keeping the old path if it is still available, maybe migrating to the new path if it responds sooner. Overall, picking an unused path ID is much cleaner. Even if the desired result is path migration, using a new path ID provides "make before break" functionality.